pylint/checkers/base.py

```diff
@@ -836,7 +836,12 @@ def _check_in_loop(self, node, node_name):
     def _check_redefinition(self, redeftype, node):
         """check for redefinition of a function / method / class name"""
         parent_frame = node.parent.frame()
-        defined_self = parent_frame[node.name]
+        # Ignore function stubs created for type information
+        defined_self = next(
+            local
```

Comment:
You might want to throw a sentinel in there with `next((....), None)` so we won't crash if we can't find a proper local.